src/frontend/src/features/rooms/api/fetchRoom.ts

```diff
@@ -0,0 +1,20 @@
+import { ApiError } from '@/api/ApiError'
+import { type ApiRoom } from './ApiRoom'
+import { fetchApi } from '@/api/fetchApi'
+
+export const fetchRoom = ({
+  roomId,
+  username = '',
+}: {
+  roomId: string
+  username?: string
+}) => {
+  return fetchApi<ApiRoom>(
+    `/rooms/${roomId}?username=${encodeURIComponent(username)}`
+  ).then((room) => {
+    if (!room.livekit?.token) {
```

Comment:
```suggestion
    if (!room?.livekit?.token) {
```

Reply:
I don't understand why this should be added as when we are here it means the api responded correctly?

Reply:
I'll merge as is and change if needed :)